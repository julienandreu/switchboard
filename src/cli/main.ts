import { pipe } from 'fp-ts/function';
import { log } from 'fp-ts/lib/Console';
import { addDescription, addVersion, createProgram } from './program';
import { MainOptions } from '../interfaces';

export const main = ({ name, version, description }: MainOptions) =>
  pipe(name, createProgram, addVersion(version), addDescription(description), log)();
