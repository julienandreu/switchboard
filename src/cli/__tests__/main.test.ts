import { main } from '../main';

describe('main', () => {
  it('should console log "Hello Switchboard"', () => {
    const consoleLogMock = jest.spyOn(console, 'log').mockImplementation();
    main();
    expect(console.log).toBeCalledWith('Hello Switchboard');
    consoleLogMock.mockRestore();
  });
});
