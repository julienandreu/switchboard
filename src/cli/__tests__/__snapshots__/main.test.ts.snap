// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should match definition and pipe flow 1`] = `[Function]`;
